test/karma.conf.js

```diff
@@ -9,6 +9,13 @@ const browserStack = {
 
 process.env.CHROME_BIN = playwright.chromium.executablePath();
 
+// BrowserStack rate limit after 1600 calls every 5 minutes.
+// Per second, https://www.browserstack.com/docs/automate/api-reference/selenium/introduction#rest-api-projects
+const MAX_REQUEST_BROWSERSTACK = 1600 / (60 * 5);
+// Estimate the max number of concurrent karma builds.
+// CircleCI accepts up to 80 concurrent builds, for each PR, 6 concurrent builds are used.
+const MAX_KARMA_CONCURRENT_BUILD = 80 / 6;
```

Comment:
The maximum amount of concurrent test_browser jobs is 80. 

I'm not following how you leap from "CircleCI accepts up to 80 concurrent builds" to "for each PR, 6 concurrent builds are used". How does a PR factor in here?

If you're estimating at some point then this should be its own magic number so that we can easily adjust later. Not baked into a well defined constant.

Seems to me we don't actually care about the CircleCI concurrency (or rather just as an upper bound for verification). Instead we're just trying to handle the maximum number of parallel test_browser jobs. I'd suggest just going with "10" for now and see how it goes.

Reply:
6 comes from the number if items in:

<img width="305" alt="Capture d’écran 2021-02-15 à 11 49 09" src="https://user-images.githubusercontent.com/3165635/107937180-ddf2ae00-6f83-11eb-97f6-45f8c205a59c.png">

Ok, let's use another magic number :)

Reply:
> 6 comes from the number if items in:

Why is the number of jobs relevant? In a review I'm not interested in the what. I can read code for myself. I need to know the reason for why did it that way because that is not obvious.

Reply:
Let's consider the following case:

- 20 open PRs at the same time.
- CircleCI can process all the first step in parallel (checkout)
- CircleCI complete the all checkout step, more or less at the same time
- CircleCI moves to the next steps in the pipelines. This time he has to handle 6 x 20 = 120 jobs, 20 of these jobs are *test_browser-1*. He picks 83 randomly. Out of the 83, 83/6 are the step *test_browser-1*.